spring-2013/hw1/src/CreditCardValidator.hs

```diff
@@ -0,0 +1,22 @@
+module CreditCardValidator
+    ( toDigits
+    , doubleEveryOther
+    , sumDigits
+    , validate
+    ) where
+
+toDigits :: Integer -> [Integer]
+toDigits n
+    | n <= 0 = []
+    | otherwise = map readChar $ show n
+    where readChar = read . (:[])
```

Comment:
I also line-break always, though I prefer the de-dented `where`

```hs
toDigits n
    | ...
    | otherwise ...
  where
    readChar = ...
```

That said, swooping up single-line sections into a tighter phrase can work well sometimes:

```hs
strip = trim . trim where trim = reverse . dropWhile isSpace
```